docs/decisions/00XX-realtime-api-clients.md

```diff
@@ -0,0 +1,206 @@
+| Content                   | Response output item done              | `response.output_item.done`                             | `-`                                       |
+| Content                   | Response content part added            | `response.content_part.added`                           | `-`                                       |
+| Content                   | Response content part done             | `response.content_part.done`                            | `-`                                       |
+| Content                   | Response text delta                    | `response.text.delta`                                   | `BidiGenerateContentServerContent`        |
+| Content                   | Response text done                     | `response.text.done`                                    | `-`                                       |
+| Content                   | Response audio transcript delta        | `response.audio_transcript.delta`                       | `BidiGenerateContentServerContent`        |
+| Content                   | Response audio transcript done         | `response.audio_transcript.done`                        | `-`                                       |
+| Content                   | Response audio delta                   | `response.audio.delta`                                  | `BidiGenerateContentServerContent`        |
+| Content                   | Response audio done                    | `response.audio.done`                                   | `-`                                       |
+| Content                   | Response function call arguments delta | `response.function_call_arguments.delta`                | `BidiGenerateContentToolCall`             |
+| Content                   | Response function call arguments done  | `response.function_call_arguments.done`                 | `-`                                       |
+| Control                   | Function call cancelled                | `-`                                                     | `BidiGenerateContentToolCallCancellation` |
+| Control                   | Rate limits updated                    | `rate_limits.updated`                                   | `-`                                       |
+
+
+## Overall Decision Drivers
+- Simple programming model that is likely able to handle future realtime api's and the evolution of the existing ones.
+- Whenever possible we transform incoming content into Semantic Kernel content, but surface everything, so it's extensible
+- Protocol agnostic, should be able to use different types of protocols under the covers, like websocket and WebRTC, without changing the client code (unless the protocol requires it), there will be slight differences in behavior depending on the protocol.
+
+There are multiple areas where we need to make decisions, these are:
+- Content and Events
+- Programming model
+- Audio speaker/microphone handling
+
+# Content and Events
```

Comment:
I don't see any section in the ADR related to testing/validation? Should we add some information about this? How do we plan to test the real-time flows?